osrdyne/src/pool.rs

```diff
@@ -477,12 +478,15 @@ async fn worker_control_loop(
     sleep_interval: Duration,
 ) {
     loop {
+        let span = span!(Level::INFO, "loop_iteration");
+        let _enter = span.enter();
```

Comment:
If you don't reuse `span` anywhere else, you should be able to use `.entered()` which produce a guard that owns the span instead of borrowing it.

```suggestion
        let _entered_guard = span!(Level::INFO, "loop_iteration").entered();
```

Reply:
Not compiling because worker_control_loop is called within a task with async move.